Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -180,7 +184,12 @@ func (s *serviceAccount) Admit(a admission.Attributes) (err error) {
 		return admission.NewForbidden(a, fmt.Errorf("service account %s/%s was not found, retry after the service account is created", a.GetNamespace(), pod.Spec.ServiceAccountName))
 	}
 
-	if s.LimitSecretReferences {
+	enforceMountableSecretCheck := false
+	if value, ok := serviceAccount.Annotations[EnforceMountableSecretsAnnotation]; ok {
+		enforceMountableSecretCheck, _ = strconv.ParseBool(value)
+	}
+
+	if s.LimitSecretReferences || enforceMountableSecretCheck {
```

Comment:
refactor to `s.EnforceMountableSecrets(sa)` with an early return and a comment explaining `LimitSecretReferences` means enforce for all, regardless of annotation. Can also just do `strconv.ParseBool(serviceAccount.Annotations[EnforceMountableSecretsAnnotation])`